controller/main.go

```diff
@@ -125,8 +125,13 @@ func (c *controller) SetPools(l log.Logger, pools *config.Pools) controllers.Syn
 		level.Error(l).Log("op", "setConfig", "error", err, "msg", "applying new configuration failed")
 		return controllers.SyncStateError
 	}
-	c.pools = pools
-	return controllers.SyncStateReprocessAll
+
+	if !reflect.DeepEqual(c.pools, pools) {
+		c.pools = pools
+		return controllers.SyncStateReprocessAll
+	}
```

Comment:
Maybe we should check the other way around, if is DeepEqual then return SyncStateSuccess,
with a log similar to what we have in the SetBalancer:
`level.Debug(l).Log("event", "noChange", "msg", "service converged, no change")`
else continue to execute the c.ips.SetPools and return Reprocess.
This is just to avoid running the c.ips.SetPools on the same/unchanged pools

Reply:
I had a reason to always run c.ips.SetPools but I can't remember what it was.

Will change it as you suggest. Thank you.

Reply:
done

Reply:
it's just a suggestion, but maybe don't negate the DeepEqual if statement,
so we won't have to branch if inside an if and make it easier to follow.
i.e. if the pools are equal we throw a log and return success, else
continue like before.

Reply:
I prefer the last return to be a success

Reply:
+1, the happy path must be the leftmost indented one unless exceptional cases.

Reply:
you can also say that the Reprocess state is the happy path, as it was previously,
anyways it is unclear to me in what scenario would we reconcile and call the pool controller handler when the pool didn't change

Reply:
> it is unclear to me in what scenario would we reconcile and call the pool controller handler when the pool didn't change

This is a valid point too. The only scenario I can think of is when we do that is because a community entry changes but is not part of a legacy addresspool, which is a pretty cornery case.

Reply:
yeah, mainly changes on community and namespaces not related to existing pools

Reply:
but the community resource is translated into a BGPadv (both legacy and non) and that's part of the pool, no?

Reply:
community affects the legacy pools. BGPadv are part of the pools indeed but in the pools controller they are always empty